polyfill/lib/calendar.mjs

```diff
@@ -422,7 +419,7 @@ impl['gregory'] = ObjectAssign({}, impl['iso8601'], {
       ['monthCode', undefined],
       ['year', undefined]
     ]);
-    if (fields.month !== undefined) gre.validateFields(fields);
+    if (fields.month !== undefined && fields.monthCode === undefined) gre.validateFields(fields);
```

Comment:
Raises the question of whether calendars should reject e.g. day = 50, or allow temporal to reject in resulting usage. The latter seems fine by me, in which case this :+1:

Reply:
Not sure I understand this comment. If day=50, it should throw in `impl['iso8601'].monthDayFromFields` before temporal core gets ahold of the result. Or were you saying something different?

Reply:
ah, thank you. I was forgetting that this raises up to the iso.monthDayFromFields too.